src/gui/test/components/navigation/footer.tsx

```diff
@@ -1,9 +1,19 @@
-import { afterEach, it, describe, expect } from 'vitest'
+import { vi, afterEach, it, describe, expect } from 'vitest'
 import { cleanup, render, screen } from '@testing-library/react'
 import html from 'diffable-html'
 import { useGraphStore as useStore } from '@/state/index.js'
 import { Footer } from '@/components/navigation/footer.jsx'
 
+vi.mock('@/components/icons/index.js', () => ({
+  Discord: 'gui-discord-icon',
+  Linkedin: 'gui-linkedin-icon',
+  Github: 'gui-github-icon',
+  TwitterX: 'gui-twitter-icon',
+}))
+vi.mock('@/components/theme-switcher/theme-switcher.jsx', () => ({
+  default: () => 'gui-theme-switcher',
```

Comment:
it shouldn't be a function, if it's just a string then it will render like a proper html tag
```suggestion
  default: 'gui-theme-switcher',
```